consul/kv_consul.go

```diff
@@ -598,6 +598,17 @@ func (kv *consulKV) LockWithTimeout(
 	}, nil
 }
 
+func (kv *consulKV) IsKeyLocked(key string) (bool, string, error) {
```

Comment:
Should we check if the provided key is indeed a lock?

Reply:
Can you file a PWX to add a functionality in consul by which we can ensure the value returned by this key is actually a lock? This doesn't exist today. We create a key without any value.

Reply:
https://portworx.atlassian.net/browse/PWX-30065